app/models/concerns/form526_rapid_ready_for_decision_concern.rb

```diff
@@ -172,21 +178,21 @@ def notify_mas_all_claims_tracking
     RrdMasNotificationMailer.build(self, Settings.rrd.mas_all_claims_tracking.recipients).deliver_now
   end
 
-  def notify_mas
+  def conditionally_notify_mas
     notify_mas_all_claims_tracking if read_metadata(:forward_to_mas_all_claims)
-
-    if read_metadata(:forward_to_mas)
-      notify_mas_tracking
-
-      if Flipper.enabled?(:rrd_mas_notification)
-        client = MailAutomation::Client.new({
-                                              file_number: birls_id,
-                                              claim_id: submitted_claim_id,
-                                              form526: form
-                                            })
-        response = client.initiate_apcas_processing
-        save_metadata(mas_packetId: response.dig('body', 'packetId'))
-      end
+    notify_mas_tracking if read_metadata(:forward_to_mas)
+
+    return unless Flipper.enabled?(:rrd_mas_notification)
+
+    if read_metadata(:forward_to_mas) ||
+       (Flipper.enabled?(:rrd_mas_all_claims_notification) && read_metadata(:forward_to_mas_all_claims))
```

Comment:
The Flipper conditional here is slightly gnarly, because it needs to ensure that whether we use the older logic or newer logic, APCAS only gets hit once. The APCAS client code itself is unchanged -- ignore whitespace changes.

Reply:
Please add some code comments to explain the conditionals. I'm glad you're planning to remove the old flags.